cmd/k8s-operator/proxygroup_test.go

```diff
@@ -517,6 +517,187 @@ func TestIngressAdvertiseServicesConfigPreserved(t *testing.T) {
+		proxyClasses         []*tsapi.ProxyClass  // ProxyClasses to create
+		defaultProxyClass    string
+		useLEStagingEndpoint bool
+	}{
+		{
+			name:                 "ingress_pg_with_staging_proxyclass",
+			pgType:               tsapi.ProxyGroupTypeIngress,
+			proxyClassName:       "le-staging",
+			proxyClasses:         []*tsapi.ProxyClass{pcLEStaging},
+			useLEStagingEndpoint: true,
+		},
+		{
+			name:                 "ingress_pg_with_other_proxyclass",
+			pgType:               tsapi.ProxyGroupTypeIngress,
+			proxyClassName:       "other",
+			proxyClasses:         []*tsapi.ProxyClass{pcOther},
+			useLEStagingEndpoint: false,
+		},
+		{
+			name:                 "ingress_pg_no_proxyclass",
+			pgType:               tsapi.ProxyGroupTypeIngress,
+			proxyClassName:       "",
+			proxyClasses:         nil,
+			useLEStagingEndpoint: false,
+		},
+		{
+			name:                 "ingress_pg_with_staging_proxyclass_false",
+			pgType:               tsapi.ProxyGroupTypeIngress,
+			proxyClassName:       "le-staging-false",
+			proxyClasses:         []*tsapi.ProxyClass{pcLEStagingFalse},
+			useLEStagingEndpoint: false,
+		},
+		{
+			name:                 "egress_pg_with_staging_proxyclass",
+			pgType:               tsapi.ProxyGroupTypeEgress,
+			proxyClassName:       "le-staging",
+			proxyClasses:         []*tsapi.ProxyClass{pcLEStaging},
+			useLEStagingEndpoint: false,
+		},
+		{
+			name:                 "ingress_pg_with_default_staging_proxyclass",
+			pgType:               tsapi.ProxyGroupTypeIngress,
+			proxyClassName:       "",
+			proxyClasses:         []*tsapi.ProxyClass{pcLEStaging},
+			defaultProxyClass:    "le-staging",
+			useLEStagingEndpoint: true,
+		},
+		{
+			name:                 "ingress_pg_with_default_other_proxyclass",
+			pgType:               tsapi.ProxyGroupTypeIngress,
+			proxyClassName:       "",
+			proxyClasses:         []*tsapi.ProxyClass{pcOther},
+			defaultProxyClass:    "other",
+			useLEStagingEndpoint: false,
+		},
+		{
+			name:                 "ingress_pg_with_default_staging_proxyclass_false",
+			pgType:               tsapi.ProxyGroupTypeIngress,
+			proxyClassName:       "",
+			defaultProxyClass:    "le-staging-false",
+			proxyClasses:         []*tsapi.ProxyClass{pcLEStagingFalse},
+			useLEStagingEndpoint: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Build fake client with scheme and ProxyClasses
+			builder := fake.NewClientBuilder().
+				WithScheme(tsapi.GlobalScheme)
+
+			// Add ProxyClass if specified and set its status
+			var pc *tsapi.ProxyClass
+			if tt.proxyClasses != nil {
+				pc = tt.proxyClasses[0]
+				builder = builder.WithObjects(pc).
+					WithStatusSubresource(pc)
+			}
+
+			fc := builder.Build()
+
+			// Set ProxyClass status if one is being used
+			if tt.proxyClassName != "" || tt.defaultProxyClass != "" {
+				pc := &tsapi.ProxyClass{}
+				name := tt.proxyClassName
+				if name == "" {
+					name = tt.defaultProxyClass
```

Comment:
I could see some value in writing a test that exercises the real default-selection logic, but this doesn't seem like it's testing any meaningfully different prod code from the other test cases. Maybe just delete this test case field unless I'm missing something?

Reply:
The test does actually test selection (i.e we trigger the reconcile loop that chooses between default proxy class as set on reconciler if any and any proxy class specified per resource)

Reply:
Oops sorry, I totally misread this 👍